application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/TaskObjectUpdateEventListener.java

```diff
@@ -0,0 +1,122 @@
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.taskmanager.internal;
+
+import java.util.Arrays;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.bridge.event.DocumentCreatingEvent;
+import org.xwiki.bridge.event.DocumentDeletingEvent;
+import org.xwiki.bridge.event.DocumentUpdatingEvent;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.observation.event.Event;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.taskmanager.TaskCounter;
+import com.xwiki.taskmanager.model.Task;
+
+/**
+ * Listener that will modify the task macro associated with the updated task object.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("TaskObjectUpdateEventListener")
+@Singleton
+public class TaskObjectUpdateEventListener extends AbstractTaskEventListener
+{
+    @Inject
+    private TaskCounter taskCounter;
+
+    /**
+     * Constructor.
+     */
+    public TaskObjectUpdateEventListener()
+    {
+        super(TaskObjectUpdateEventListener.class.getName(), Arrays.asList(new DocumentUpdatingEvent(),
+            new DocumentCreatingEvent(), new DocumentDeletingEvent()));
+    }
+
+    @Override
+    protected void processEvent(XWikiDocument document, XWikiContext context, Event event)
+    {
+        if (event instanceof DocumentDeletingEvent) {
+            try {
+                XWikiDocument actualDoc = context.getWiki().getDocument(document.getDocumentReference(), context);
+                BaseObject object = actualDoc.getXObject(TASK_CLASS_REFERENCE);
+                if (object != null && !object.getStringValue(Task.OWNER).isEmpty()) {
+                    taskXDOMProcessor.removeTaskMacroCall(document.getDocumentReference(),
+                        resolver.resolve(object.getStringValue(Task.OWNER)), context);
+                }
+            } catch (XWikiException e) {
+                logger.warn("Failed to remove the macro call from the owner document of the task [{}]",
+                    document.getDocumentReference());
+            }
+            return;
+        }
+        BaseObject taskObj = document.getXObject(TASK_CLASS_REFERENCE);
+
+        if (taskObj == null) {
+            return;
+        }
+
+        setTaskNumber(context, taskObj);
+        taskObj.set(Task.RENDER,
+            taskXDOMProcessor.renderTaskByReference(taskObj.getDocumentReference(), document.getSyntax()), context);
+
+        if (context.get(TASK_UPDATE_FLAG) != null || taskObj.getStringValue(Task.OWNER).isEmpty()) {
+            return;
+        }
+
+
+        String taskOwner = taskObj.getStringValue(Task.OWNER);
+
+        DocumentReference taskOwnerRef = resolver.resolve(taskOwner, EntityType.DOCUMENT);
```

Comment:
```suggestion
        DocumentReference taskOwnerRef = resolver.resolve(taskOwner, EntityType.DOCUMENT, document.getDocumentReference());
```
The owner reference needs to be resolved relative to the task reference.